packages/cli/src/lifecycles/graph.js

```diff
@@ -76,6 +76,16 @@ const createGraphFromPages = async (pagesDir, config) => {
                 // set <title></title> element text, override with markdown title
                 title = title || '';
 
+                // create webpack chunk name based on route and page name
+                const routes = route.lastIndexOf('/') === route.length - 1 && route.lastIndexOf('/') > 0
```

Comment:
what is this conditional expressing?  Maybe better to put it into a variable to improve readability?

Reply:
becausewedwouldhavevariablenamesthislong

Reply:
I only asked a question 🤷

Reply:
I mean I originally wrote that 6 months ago and was still able to understand it fairly quickly.

Reply:
Yeah, I'm just trying to think more of how this would be read / understood by someone else coming to the project fresh who hasn't spent a year+ working on it, or how you would explain the code to someone else as you walked through it.  Length or mild verboseness shouldn't really be considered as some sort of gatekeeping metric, to be fair, but rather the value it adds to the overall expressiveness of the code.  Mapping logic to context (variables) can thus help reinforce the intent / readability of the code so much so that it almost comes out as natural language (or at least is always my goal).

When the code itself reads like a book, then you never really need comments, unless it's maybe capture the why (specific business rule / case, environment constraint, etc), as opposed to the what.

Another factor to my comment is also more that we have a lot of similar logic like this around our code [[1](https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/src/lifecycles/graph.js#L45)], [[2](https://github.com/ProjectEvergreen/greenwood/blob/master/packages/cli/src/lifecycles/scaffold.js#L45)], like in our resolvers and serialize code that I think we have come to accept over time.  I have a card in Trello where I had wanted to make a more technical document for ourselves and contributing developers that would explain the architecture of the project, and how it is organized overall.  I think a key outcome of that exercise will be a better understood taxonomy (or in Domain Driven Design, they would call it a ubiquitous language) which would then allow us to better express these (at first glance) arbitrary bits of logic, and / or could lead to a more refined implementation of Greenwood itself as we get more awareness of what we're trying to achieve.

I'll add this convo as notes to that card.

Reply:
hah.  there was already a note for this exact topic in the card already 🤓